tests/integration/test_authd/test_authd_key_request_func.py

```diff
@@ -0,0 +1,151 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check if the 'wazuh-authd' daemon correctly handles the key requests
+       from agents with pre-existing IP addresses or IDs.
+
+tier: 0
+
+modules:
+    - authd
+
+components:
+    - manager
+
+daemons:
+    - wazuh-authd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html
+    - https://documentation.wazuh.com/current/user-manual/registering/key-request.html
+
+tags:
+    - key_request
+'''
+import os
+import re
+from pathlib import Path
+
+import pytest
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH, WAZUH_PATH
+from wazuh_testing.utils.configuration import load_configuration_template, get_test_cases_data
+from wazuh_testing.tools.monitors.file_monitor import FileMonitor
+from wazuh_testing.utils import callbacks
+from wazuh_testing.modules.authd import PREFIX
+
+from . import CONFIGURATIONS_FOLDER_PATH, TEST_CASES_FOLDER_PATH
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Test configurations
+local_internal_options = {'authd.debug': '2'}
+
+# Configurations
+test_configuration_path = Path(CONFIGURATIONS_FOLDER_PATH, 'config_authd_key_request_func.yaml')
+test_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_authd_key_request_func.yaml')
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+
+# Variables
+script_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'files')
+script_filename = 'fetch_keys.py'
+kreq_sock_path = os.path.join(WAZUH_PATH, 'queue', 'sockets', 'krequest')
```

Comment:
Add this queue to the framework in `constants/paths/sockets.py` and use it. Modify this where needed.